File: sql/plsql/TypeScript/PlSqlLexerBase.ts
import { CommonToken, Lexer, CharStream, Token } from "antlr4";
import PlSqlParser from './PlSqlParser';

export default abstract class PlSqlLexerBase extends Lexer {
  self : PlSqlLexerBase;
  
  IsNewlineAtPos(pos: number): boolean {
    const la = this._input.LA(pos);
    return la == -1 || String.fromCharCode(la) == '\n';
  }

}

File: sql/plsql/Dart/PlSqlLexerBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:convert';

abstract class PlSqlLexerBase extends Lexer
{
    PlSqlLexerBase(CharStream input) : super(input)
    {
    }

    bool IsNewlineAtPos(int pos)
    {
        int la = inputStream.LA(pos)!;
		if (la == -1) return true;
		return '\n' == String.fromCharCode(inputStream.LA(pos)!);
    }
}
